test/plugin/run.sh

```diff
@@ -151,6 +151,11 @@ agent_home_selector() {
 start_stamp=`date +%s`
 parse_commandline "$@"
 
+if [[ $(uname) == 'Darwin' ]] && [[ $(uname -m) == 'arm64' ]]; then
+  nohup socat TCP-LISTEN:2375,range=127.0.0.1/32,reuseaddr,fork UNIX-CLIENT:/var/run/docker.sock > /dev/null 2>&1 &
```

Comment:
Can you add a `trap` to capture the exit signal and kill this process so when the tests finished this can be released? Otherwise this process is still running even when the tests finished.

Reply:
> From the document, https://github.com/fabric8io/docker-maven-plugin/blob/master/doc/changelog.md
> 
> The M1 support has already been added from `0.38.1` and I've run this patch without `socat`, everything is fine.
> 
> ![image](https://user-images.githubusercontent.com/2568208/199633675-1045ddbd-c4b6-484f-a825-b501dd8ef30e.png)

What version of mac m1 docker desktop are you using?

Reply:
I use the newest version of m1 docker desktop, still error if not use socat

```
[INFO] 
[INFO] --- docker-maven-plugin:0.40.2:build (build) @ tomcat-container ---
./test/plugin/run.sh: line 186: 50210 Killed: 9               ${mvnw} --batch-mode -f ${home}/pom.xml -DskipTests -Dbase_image_java=${base_image_java} -Dbase_image_tomcat=${base_image_tomcat} -Dcontainer_image_version=${container_image_version} clean package

```

Reply:
> Can you add a `trap` to capture the exit signal and kill this process so when the tests finished this can be released? Otherwise this process is still running even when the tests finished.

I have removed socat nohup, when the run.sh exit, socat exit.

Reply:
I've confirmed with another laptop (13' M2 2022). Previously I've checked on my Mac Studio with M1 Ultra. Still the same results. 

```shell
$ bash ./test/plugin/run.sh -f spring-4.3.x-scenario
```

<img width="1440" alt="image" src="https://user-images.githubusercontent.com/2568208/200338878-3c66028f-b866-4cf4-be26-e6d933843765.png">

I'm using the default configuration with Docker Desktop `4.13.1 (90346)`,

<img width="1381" alt="image" src="https://user-images.githubusercontent.com/2568208/200339321-f51f931d-26e4-47bb-b6e2-b590234f03f4.png">

I guess your jvm may not be built for arm64. Anyway, I don't think `socat` is necessary.

Reply:
@lujiajing1126 Could you submit another PR to bump up the version? 
Then I could do another round check locally on my M1 Max.

Reply:
> @lujiajing1126 Could you submit another PR to bump up the version? Then I could do another round check locally on my M1 Max.

Sure